Week 8/scheduler/src/Scheduler.java

```diff
@@ -0,0 +1,95 @@
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.util.PriorityQueue;
+
+// Schedule the jobs in the given file and prints the log.
+// The scheduler selects tasks using priority queue.
+// Information on job priority is contained in Job class.
+public class Scheduler {
+  private PriorityQueue<Job> jobPriorityQueue = new PriorityQueue<>();
+  private int time = 0;
+  private int[] endtime; // An array to save the end time of jobs.
+
+  public Scheduler(String inputPath) throws Exception {
+    setJobPriorityQueue(inputPath);
+  }
+
+  // Read the input file and add the jobs in the priority queue one by one.
+  public void setJobPriorityQueue(String inputPath) throws Exception {
+    FileInputStream inputStream = null;
+    inputStream = new FileInputStream(inputPath);
+    BufferedReader inFile = new BufferedReader(new InputStreamReader(inputStream));
+
+    // Get the number of jobs.
+    int numberOfJobs = Integer.parseInt(inFile.readLine());
+
+    // Initialize the endtime array.
+    endtime = new int[numberOfJobs];
+
+    // Insert jobs in the input file in the priority queue.
+    insertJobs(numberOfJobs, inFile);
+  }
```

Comment:
Handle the exception from file read in this function.
also if file read fails return false
if it worked return true

Reply:
Also don't handle exception as Exception.
You specify which exception is it